specification/cognitiveservices/OpenAI.Inference/routes.tsp

```diff
@@ -41,10 +43,38 @@ Gets chat completions for the provided chat messages.
 Completions support a wide variety of tasks and generate text that continues from or "completes"
 provided prompt data.
 """)
-@TypeSpec.Rest.actionSeparator("/")
+@added(ServiceApiVersions.v2023_05_15)
+@actionSeparator("/")
 @action("chat/completions") //@convenientAPI(true)
-op getChatCompletions is Azure.Core.ResourceAction<
+op getChatCompletions is ResourceAction<
   Deployment,
   ChatCompletionsOptions,
   ChatCompletions
 >;
+
+#suppress "@azure-tools/typespec-azure-core/no-rpc-path-params" "Allowed because this is a non-standard status polling operation."
+@doc("Returns the status of the images operation")
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@route("/operations/images/{operationId}")
+op getImageOperationStatus is RpcOperation<
+  {
+    @doc(".") @path operationId: string;
+  },
+  ImageOperationResponse
+>;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
+@doc("Starts the generation of a batch of images from a text caption")
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@route("/images/generations:submit")
+@pollingOperation(
+  getImageOperationStatus,
+  {
+    operationId: ResponseProperty<"id">,
+  }
+)
+op startGenerateImage is OaiLongRunningRpcOperation<
+  ImageGenerationOptions,
+  ImageOperationResponse,
+  ImageOperationStatus
+>;
```

Comment:
Do we have a runtime log on the LRO responses? Or at least give us a response of last LRO poll.

Ask because the LRO may not be correctly defined.

It may need to be written as
```
op startGenerateImage is OaiLongRunningRpcOperation<
  ImageGenerationOptions,
  ImageOperationResponse,
  ImageResponse
>;
```

The `TStatusResult` in template is the model in "result" property. It is a **result**, not a **status**.
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/foundations.tsp#L37-L51

Reply:
Or you may want to use [Azure.Core.LongRunningRpcOperation](https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/operations.tsp#L37-L63)